sdk/src/wallet/operations/transaction/prepare_transaction.rs

```diff
@@ -37,56 +32,13 @@ where
             output.verify_storage_deposit(storage_score_params)?;
         }
 
-        if let Some(custom_inputs) = options.as_ref().and_then(|options| options.custom_inputs.as_ref()) {
-            // validate inputs amount
-            if !INPUT_COUNT_RANGE.contains(&(custom_inputs.len() as u16)) {
-                return Err(crate::types::block::Error::InvalidInputCount(
-                    TryIntoBoundedU16Error::Truncated(custom_inputs.len()),
-                ))?;
-            }
-        }
-
-        if let Some(mandatory_inputs) = options.as_ref().and_then(|options| options.mandatory_inputs.as_ref()) {
-            // validate inputs amount
-            if !INPUT_COUNT_RANGE.contains(&(mandatory_inputs.len() as u16)) {
-                return Err(crate::types::block::Error::InvalidInputCount(
-                    TryIntoBoundedU16Error::Truncated(mandatory_inputs.len()),
-                ))?;
-            }
+        if options.required_inputs.len() as u16 > INPUT_COUNT_MAX {
+            return Err(crate::types::block::Error::InvalidInputCount(
+                TryIntoBoundedU16Error::Truncated(options.required_inputs.len()),
+            ))?;
         }
 
-        let remainder_address = options
-            .as_ref()
-            .and_then(|options| match &options.remainder_value_strategy {
-                RemainderValueStrategy::ReuseAddress => None,
-                RemainderValueStrategy::CustomAddress(address) => Some(address.clone()),
-            });
-
-        let selected_transaction_data = self
-            .select_inputs(
-                outputs,
-                options
-                    .as_ref()
-                    .and_then(|options| options.custom_inputs.as_ref())
-                    .map(|inputs| HashSet::from_iter(inputs.clone())),
-                options
-                    .as_ref()
-                    .and_then(|options| options.mandatory_inputs.as_ref())
-                    .map(|inputs| HashSet::from_iter(inputs.clone())),
-                remainder_address,
-                options.as_ref().and_then(|options| options.burn.as_ref()),
-                options.as_ref().and_then(|options| options.mana_allotments.clone()),
-            )
-            .await?;
-
-        let prepared_transaction_data = match self.build_transaction(selected_transaction_data.clone(), options).await {
-            Ok(res) => res,
-            Err(err) => {
-                // unlock outputs so they are available for a new transaction
-                self.unlock_inputs(&selected_transaction_data.inputs).await?;
```

Comment:
Has this been moved somewhere else?

Reply:
no, it's not needed anymore because we wait to lock until after building the prepared txn succeeds